web-app/django/VIM/apps/instruments/management/commands/import_instruments.py

```diff
@@ -0,0 +1,122 @@
+import csv
+from typing import Optional
+import requests
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from VIM.apps.instruments.models import Instrument, InstrumentName, Language, AVResource
+
+
+class Command(BaseCommand):
+    """
+    The import_instruments command imports instrument objects from Wikidata.
+
+    NOTE: For now, this script only imports instrument names in English and French. It
+    also only imports a set of previously-curated instruments that have images available.
+    This list of instruments is stored in startup_data/vim_instruments_with_images-15sept.csv
+    """
+
+    help = "Imports instrument objects"
+
+    def parse_instrument_data(
+        self, instrument_id: str, instrument_data: dict
+    ) -> dict[str, str | dict[str, str]]:
+        """
+        Given a dictionary response from the wbgetentities API, parse the data into a
+        dictionary of desired instrument data.
+
+        instrument_id [str]: Wikidata ID of the instrument
+        instrument_data [dict]: Dictionary response from wbgetentities API
+
+        return [dict]: Dictionary of parsed instrument data, containing the following
+            keys:
+            - wikidata_id [str]: Wikidata ID of the instrument
+            - ins_names [dict]: Dictionary of instrument names, with language codes as
+                keys and instrument names as values
+            - hornbostel_sachs_class [str]: Hornbostel-Sachs classification of the
+                instrument
+            - mimo_class [str]: MIMO classification of the instrument
+        """
+        # Get available instrument names
+        ins_labels: dict = instrument_data["labels"]
+        ins_names: dict[str, str] = {
+            value["language"]: value["value"] for key, value in ins_labels.items()
+        }
+        # Get Hornbostel-Sachs and MIMO classifications, if available
+        ins_hbs: Optional[list[dict]] = instrument_data["claims"].get("P1762")
+        ins_mimo: Optional[list[dict]] = instrument_data["claims"].get("P3763")
+        if ins_hbs and ins_hbs[0]["mainsnak"]["snaktype"] == "value":
+            hbs_class: str = ins_hbs[0]["mainsnak"]["datavalue"]["value"]
+        else:
+            hbs_class = ""
+        if ins_mimo and ins_mimo[0]["mainsnak"]["snaktype"] == "value":
+            mimo_class: str = ins_mimo[0]["mainsnak"]["datavalue"]["value"]
```

Comment:
I, as well as several others in the lab currently, are bemused and intrigued by wikidata's use of the word "snak"...

(no action required here, I assume that the relevant values are contained in the indicated locations - we just found this interesting)